tests/virt_autotest/kubevirt_tests_server.pm

```diff
@@ -264,33 +296,159 @@ sub deploy_kubevirt_manifests {
+    record_info('Apply test config', '');
+
+    # bsc#1210696, 1210863
+    our $local_registry_ip;
+    assert_script_run("cat > 1210696_1210863.yaml <<EOF
+spec:
+  config:
+    dataVolumeTTLSeconds: -1
+    featureGates:
+    - HonorWaitForFirstConsumer
+    insecureRegistries:
+    - registry:5000
+    - fakeregistry:5000
+    - $local_registry_ip:5000
+    uploadProxyURLOverride: https://127.0.0.1:31001
+EOF
+(exit \$?)");
+    assert_script_run("kubectl patch cdi cdi --type merge --patch-file 1210696_1210863.yaml");
+
+    # bsc#1210856
+    assert_script_run("mkdir -p /var/provision/kubevirt.io/tests && mount -t tmpfs tmpfs /var/provision/kubevirt.io/tests");
+    assert_script_run("echo 'tmpfs /var/provision/kubevirt.io/tests tmpfs rw 0 1' >> /etc/fstab");
+
+    # bsc#1210884
+    assert_script_run(qq(kubectl -n kubevirt patch kubevirt kubevirt --type merge --patch '{"spec": {"configuration": {"developerConfiguration": {"pvcTolerateLessSpaceUpToPercent": 20}}}}'));
+
+    # bsc#1210906
+    assert_script_run("sysctl -w vm.unprivileged_userfaultfd=1");
+    # Installing Whereabouts plugin
+    assert_script_run("git clone https://github.com/k8snetworkplumbingwg/whereabouts && cd whereabouts", 600);
+    assert_script_run("kubectl apply -f doc/crds/daemonset-install.yaml " .
+          "-f doc/crds/whereabouts.cni.cncf.io_ippools.yaml " .
+          "-f doc/crds/whereabouts.cni.cncf.io_overlappingrangeipreservations.yaml && cd");
+}
 
-    if (is_transactional) {
-        transactional::exit_trup_shell_and_reboot();
-        assert_script_run('mkdir -p /root/tmp && mount -o bind /root/tmp /mnt');
+sub run_virt_tests {
+    my $self = shift;
+    my $test_suite_conf;
+    my $test_scope_conf;
+    my @kubevirt_tests;
+    our $kubevirt_ver;
+
+    record_info('Run kubevirt tests', $kubevirt_ver);
+
+    # Create a valid config for the test suite
+    # Default testing config path: /usr/share/kube-virt/manifests/testing/default-config.json
+    if ($kubevirt_ver lt "0.50.0") {
+        $test_suite_conf = '/tmp/local-config.json';
+        assert_script_run(qq(cat > $test_suite_conf <<EOF
+{
+    "storageClassLocal":       "local-path",
+    "storageClassHostPath":    "host-path",
+    "storageClassHostPathSeparateDevice":    "host-path-sd",
+    "storageClassBlockVolume": "block-volume",
+    "storageClassRhel":        "rhel",
+    "storageClassWindows":     "windows",
+    "manageStorageClasses":     true
+}
+EOF
+(exit \$?)));
+    } else {
+        $test_suite_conf = '/tmp/longhorn-config.json';
+        assert_script_run(qq(cat > $test_suite_conf <<EOF
+{
+    "storageClassRhel":        "longhorn-default",
+    "storageClassWindows":     "longhorn-default",
+    "storageRWXFileSystem":    "longhorn-default",
+    "storageRWXBlock":         "longhorn-migratable",
+    "storageRWOFileSystem":    "longhorn-wffc",
+    "storageRWOBlock":         "longhorn-migratable",
+    "storageSnapshot":         "longhorn-default"
```

Comment:
From the left configured column values, it can be seen that  /usr/share/kube-virt/manifests/testing/default-config.json should be evolving too. Hardcoding the $test_suite_conf contents may loose the native changes of the  file, eg some new configure items.  Can it be refined a little bit to gain higher survival chance?

Reply:
Actually, there are only below contents from the default config file. Current configs are based on dev CI code, we have to sync with dev changes. 
```
{
    "storageClassRhel":        "rhel",
    "storageClassWindows":     "windows",
    "storageRWOFileSystem":    "local"
}
```